docs/data/data-grid/server-side-data/row-grouping.md

```diff
@@ -86,6 +86,19 @@ Server-side row grouping does not support the [`colDef.groupingValueGetter()`](/
 Use `dataSource.getGroupKey()` to compute the group key for the row instead.
 :::
 
+## Row expansion state persistence
+
+The Data Grid compares the IDs of the newly fetched rows with the existing tree level.
+Matching rows are not recreated and they maintain their expansion state and child rows.
+
+To drop all child rows and reset the expansion state, pass `collapseChildren` flag to the fetch options of the `dataSource.fetchRows()` API method.
```

Comment:
Nitpick.

I think `keepChildrenExpanded` (defaulting to `true`) communicates more clearly that “children which were already expanded, if included in new response, will expanded again”

`collapseChildren` would make more sense for client-side data, but in a server-side context with loading states, users already understand that data is being re-fetched. In that case, `collapseChildren` might be a bit confusing.